components/Map/Map.js

```diff
@@ -3,18 +3,47 @@ import "leaflet/dist/leaflet.css";
 import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
 import "leaflet-defaulticon-compatibility";
 import styled from "styled-components";
+import { useRouter } from "next/router";
 
 import LocationMarker from "./LocationMarker";
 import MyPopup from "../MyPopup";
+import ConfirmPopup from "../ConfirmPopup";
+import Click from "./Click";
 import { lostIcon, foundIcon, finishedIcon, locationIcon } from "@/lib/icons";
+import { useState } from "react";
 
-export default function Map({ items }) {
+export default function Map({ items, clickPosition, onPosition }) {
+  const router = useRouter();
+
+  const [isConfirmPopupOpen, setIsConfirmPopupOpen] = useState(false);
+
+  function handleConfirm() {
+    setIsConfirmPopupOpen(false);
+    router.push("/create");
+  }
+
+  function handleClose() {
+    setIsConfirmPopupOpen(false);
+  }
+
+  console.log(isConfirmPopupOpen);
   return (
     <StyledMapContainer
       center={[52.518623, 13.388517]}
       zoom={12}
       scrollWheelZoom
     >
+      <Click
+        onPosition={onPosition}
+        setIsConfirmPopupOpen={setIsConfirmPopupOpen}
+      />
+
+      {isConfirmPopupOpen && (
+        <Popup position={clickPosition} closeButton={false}>
+          <ConfirmPopup onConfirm={handleConfirm} onClose={handleClose} />
+        </Popup>
+      )}
+
```

Comment:
We think this solution is a little bit complicated and you could handle the Popup like in the following example: https://github.com/doemser/nextjs-react-leaflet-map/blob/main/components/Map.jsx#L36-L54
The logic of the popup should be tied to the click and not split into two components that cannot function independently from each other.

Reply:
Hi Mareike, 
thank you for reviewing and your tips!